test/github-issues/3158/entity/SessionSettings.ts

```diff
@@ -1,12 +1,15 @@
-import { Entity, OneToOne, JoinColumn } from "../../../../src";
+import { Entity, OneToOne, JoinColumn, PrimaryColumn } from "../../../../src";
 import { Session } from "./Session";
 
 @Entity({
     name: "SessionSettings"
 })
 export class SessionSettings  {
 
-    @OneToOne(type => Session, session => session.id, { primary: true})
+    @PrimaryColumn()
+    id: number;
```

Comment:
it could called `sessionId`, JFYI

Reply:
You're right. It would be much more readable if it's named like that.